src/ui/articlewebview.cc

```diff
@@ -13,167 +13,170 @@
 #include <qt_windows.h>
 #endif
 
-ArticleWebView::ArticleWebView( QWidget *parent ):
+ArticleWebView::ArticleWebView( QWidget * parent ):
   QWebEngineView( parent ),
   midButtonPressed( false ),
   selectionBySingleClick( false )
 {
   auto page = new ArticleWebPage( this );
   connect( page, &ArticleWebPage::linkClicked, this, &ArticleWebView::linkClicked );
   this->setPage( page );
+  page->deleteLater();
```

Comment:
no,I'm not sure about this.

But this line can really eliminate the warning .

put this on the destructor has no effect, the warning still present.

Reply:
I think https://github.com/xiaoyifang/goldendict-ng/issues/955#issuecomment-1631197741 is because the page is gone 😅 

---

We actually never set a profile for the articlewebpage. Qt manages the default profile (This is somehow not our fault).

https://doc.qt.io/qt-6/qwebenginepage.html#QWebEnginePage

https://github.com/xiaoyifang/goldendict-ng/blob/9c1413adeaa02d59d745d052d391fa32d4d4374c/src/ui/articlewebpage.cc#L4-L7

Code like this doesn't have this warning

https://github.com/shenlebantongying/kpDict/blob/6cc58cc569b3bfcc5e6879c1704274eea0b6fa31/src/kpWebView.cpp#L5-L11

A correct fix to surpass the warning is to create a QWebEngineProfile, instead of modifying the defaultProfile. Then pass the created profile to the ArticleWebView, which then passes it to pages (initialize with this <https://doc.qt.io/qt-6/qwebenginepage.html#QWebEnginePage-1>).

By doing so, we will manage the lifetime of the profile by ourselves.

The code below should be replaced with `auto * webProfile = new QWebEngineProfile(this, aka mainwindow)` -> `webProfile.installUrlHandler(....)`. The webProfile will be passed to the webview/webpage.

https://github.com/xiaoyifang/goldendict-ng/blob/9c1413adeaa02d59d745d052d391fa32d4d4374c/src/ui/mainwindow.cc#L161-L180

Reply:
I think to eliminate the warning ,QWebEnginePage must be release before QWebEngineProfile.

Reply:
Yes, we should create a profile and manage it on our side instead of relying on the default one which yield a warning by itself 😅

Reply:
> Yes, we should create a profile and manage it on our side instead of relying on the default one which yield a warning by itself 😅

tried this method,still has this warning.

Reply:
![image](https://github.com/xiaoyifang/goldendict-ng/assets/105986/54da13bf-ca1f-4ced-af11-90308766b086)

QWebEngineProfile is also a QObject.
```
QWebEngineProfile profile;
QWebEnginePage(profile)  ;   //which constructor will be called.
```

Reply:
C++ guarantees that "objects get destroyed in the reverse order they appear in the class declaration." https://learn.microsoft.com/en-us/cpp/cpp/destructors-cpp?view=msvc-170#order-of-destruction

Could this be why the custom webengine profile got deleted earlier than the articleview?

Maybe we can manually delete the articleview (which has pages as children) in mainwindow's destructor so that the profile will be guaranteed to be deleted after webview/pages.

Reply:
> The scanpopup's articleview is not deleted here btw. (Wait, did I introduce this warning? 😅)

The answer yes.

Adding this below https://github.com/xiaoyifang/goldendict-ng/blame/26e7ab6baabc282af72eaa169d149b34567f7e1a/src/ui/mainwindow.cc#L1109-L1117
```
delete scanPopup;
```
will shut down the warning. The scanpopup's articleview is not deleted before the deletion of the profile.

Surprise `:)`

Reply:
I'll try this later

Reply:
> > The scanpopup's articleview is not deleted here btw. (Wait, did I introduce this warning? 😅)
> 
> Adding this below https://github.com/xiaoyifang/goldendict-ng/blame/26e7ab6baabc282af72eaa169d149b34567f7e1a/src/ui/mainwindow.cc#L1109-L1117
> 
> ```
> delete scanPopup;
> ```
> 
> will shut down the warning. The scanpopup's articleview is not deleted before the deletion of the profile.
> 
> Surprise `:)`

This works.

What about define scanPopup with `parent ` or use a unique_ptr

Reply:
> What about define scanPopup with parent

scanPopup & mainwindow are two different & independent windows. I don't think we should change that. scanPopup's parent has been 0/nullptr since the very beginning of GD.

> unique_ptr

unique_ptr does not change the order of deletion. I think just adding delete is good enough 😅

Reply:
> scanPopup & mainwindow are two different & independent windows.

in reality ,the scanPopup was declared as a variable of mainWindow

Reply:
> unique_ptr does not change the order of deletion. I think just adding delete is good enough 😅

So let it be so for now.